launcher/src/main/java/org/apache/spark/launcher/SparkClassCommandBuilder.java

```diff
@@ -108,7 +108,7 @@
     }
 
     String mem = firstNonEmpty(memKey != null ? System.getenv(memKey) : null, DEFAULT_MEM);
-    cmd.add("-Xmx" + mem);
+    cmd.add("-Xmx" + addDefaultMSuffixIfNeeded(mem));
```

Comment:
we should update the standalone docs as well for --memory (SPARK_WORKER_MEMORY and SPARK_DAEMON_MEMORY) to say default to m and ideally make consistent with above docs.

Reply:
Note we should test those as well if you haven't already

Reply:
Thanks I really focused on XMX and XMS settings but now I see there another error at
https://github.com/apache/spark/blob/cc495c1c45ac0648156b662fdc308287c79f3fdc/core/src/main/scala/org/apache/spark/memory/UnifiedMemoryManager.scala#L225